pandas/tests/reshape/test_from_dummies.py

```diff
@@ -0,0 +1,321 @@
+            "col1_a": [1, 0, 0],
+            "col1_b": [0, 1, 0],
+            "col2_a": [0, 1, 0],
+            "col2_b": [0, 0, 0],
+            "col2_c": [0, 0, 1],
+        },
+    )
+
+
+def test_from_dummies_to_series_basic():
+    dummies = DataFrame({"a": [1, 0, 0, 1], "b": [0, 1, 0, 0], "c": [0, 0, 1, 0]})
+    expected = Series(list("abca"))
+    result = from_dummies(dummies, to_series=True)
+    assert all(result == expected)
+
+
+def test_from_dummies_to_series_dummy_na():
+    dummies = DataFrame({"a": [1, 0, 0], "b": [0, 1, 0], "NaN": [0, 0, 1]})
+    expected = Series(["a", "b", np.nan])
+    result = from_dummies(dummies, to_series=True, dummy_na=True)
+    assert all(result[:2] == expected[:2])
+    assert result[2] is expected[2]
+
+
+def test_from_dummies_to_series_contains_nan():
+    dummies = DataFrame({"a": [1, 0, 0], "b": [0, 1, 0]})
+    expected = Series(["a", "b", np.nan])
+    result = from_dummies(dummies, to_series=True)
+    assert all(result[:2] == expected[:2])
+    assert result[2] is expected[2]
+
+
+def test_from_dummies_to_series_dropped_first():
+    dummies = DataFrame({"a": [1, 0, 0], "b": [0, 1, 0]})
+    expected = Series(["a", "b", "c"])
+    result = from_dummies(dummies, to_series=True, dropped_first="c")
+    assert all(result == expected)
+
+
+def test_from_dummies_to_series_wrong_dropped_first():
+    dummies = DataFrame({"a": [1, 0, 1], "b": [0, 1, 1]})
+    with pytest.raises(
+        ValueError,
+        match=r"Only one dropped first value possible in 1D dummy DataFrame.",
+    ):
+        from_dummies(dummies, to_series=True, dropped_first=["c", "d"])
+
+
+def test_from_dummies_to_series_multi_assignment():
+    dummies = DataFrame({"a": [1, 0, 1], "b": [0, 1, 1]})
+    with pytest.raises(
+        ValueError, match=r"Dummy DataFrame contains multi-assignment in row 2."
+    ):
+        from_dummies(dummies, to_series=True)
+
+
+def test_from_dummies_to_series_unassigned_row():
+    dummies = DataFrame({"a": [1, 0, 0], "b": [0, 1, 0]})
+    with pytest.raises(
+        ValueError, match=r"Dummy DataFrame contains no assignment in row 2."
+    ):
+        from_dummies(dummies, to_series=True, dummy_na=True)
+
+
+def test_from_dummies_no_dummies():
+    dummies = DataFrame(
+        {"a": [1, 6, 3, 1], "b": [0, 1, 0, 2], "c": ["c1", "c2", "c3", "c4"]}
+    )
+    expected = dummies
+    result = from_dummies(dummies)
+    assert all(result == expected)
+
+
+def test_from_dummies_to_df_basic(dummies_basic):
+    expected = DataFrame(
+        {"C": [1, 2, 3], "col1": ["a", "b", "a"], "col2": ["b", "a", "c"]}
+    )
+    result = from_dummies(dummies_basic)
+    assert all(result == expected)
+
+
+def test_from_dummies_to_df_variable_string(dummies_basic):
+    expected = DataFrame(
+        {"C": [1, 2, 3], "varname0": ["a", "b", "a"], "varname1": ["b", "a", "c"]}
+    )
+    result = from_dummies(dummies_basic, variables="varname")
+    assert all(result == expected)
```

Comment:
not convinced about having `variables` as an argument - what would be a use-case? Can this achieved in the same way using `from_dummies(dummies_basic).rename(`? If so, I'd avoid the extra complexity

Reply:
The the use-case would be someone having dummies with generic prefixes and the actual variable names in an extra document. But you are right, it seems to be a much cleaner approach for the function itself as well in the actual use-case. I think its easier to read and follow in the "two steps" approach.